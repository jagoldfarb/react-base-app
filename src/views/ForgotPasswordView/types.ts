export interface IForgotPasswordFormData {
  email: string;
}

export interface IForgotPasswordViewProps {
  onSubmit: (formData: IForgotPasswordFormData) => void;
}
