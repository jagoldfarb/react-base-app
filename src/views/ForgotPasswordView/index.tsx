import React, { FunctionComponent } from 'react';
import i18next from 'i18next';

import ForgotPasswordForm from 'views/ForgotPasswordView/components/ForgotPasswordForm';

import { IForgotPasswordViewProps } from './types';
import {
  ForgotPasswordContainer,
  ForgotPasswordContent,
  Title,
  Description,
  ForgotPasswordFormContainer,
  CreateAccountContainer,
  CreateAccountLink,
} from './styles';
import './i18n';

const ForgotPasswordView: FunctionComponent<IForgotPasswordViewProps> = (
  props: IForgotPasswordViewProps,
) => {
  const { onSubmit } = props;

  return (
    <ForgotPasswordContainer data-testid="forgot-password-view">
      <ForgotPasswordContent>
        <Title>{i18next.t('FORGOT_PASSWORD_VIEW:TITLE')}</Title>
        <Description>{i18next.t('FORGOT_PASSWORD_VIEW:DESCRIPTION')}</Description>
        <ForgotPasswordFormContainer>
          <ForgotPasswordForm onSubmit={onSubmit} />
        </ForgotPasswordFormContainer>
        <CreateAccountContainer>
          <CreateAccountLink to="/signup">
            {i18next.t('FORGOT_PASSWORD_VIEW:CREATE_ACCOUNT')}
          </CreateAccountLink>
        </CreateAccountContainer>
      </ForgotPasswordContent>
    </ForgotPasswordContainer>
  );
};

export default ForgotPasswordView;
