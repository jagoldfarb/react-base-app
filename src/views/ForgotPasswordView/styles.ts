import styled from 'styled-components';
import { Link } from 'react-router-dom';

import { COLORS } from 'helpers/enums/colors';

export const ForgotPasswordContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const ForgotPasswordContent = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.h1`
  font-size: 32px;
  font-weight: bold;
  margin: 0;
`;

export const Description = styled.span`
  font-size: 16px;
  margin-top: 30px;
`;

export const ForgotPasswordFormContainer = styled.div`
  margin-top: 30px;
`;

export const CreateAccountContainer = styled.div`
  width: 100%;
  font-size: 14px;
  line-height: 1.5;
  text-align: center;
`;

export const CreateAccountLink = styled(Link)`
  color: ${COLORS.greenChristi};
  margin-left: 5px;

  :hover {
    color: ${COLORS.gray};
  }
`;
