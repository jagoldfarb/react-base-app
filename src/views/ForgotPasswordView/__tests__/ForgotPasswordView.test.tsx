import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import store from 'redux/store';

import ForgotPasswordView from '..';

const defaultProps = {
  onSubmit: jest.fn(),
};

describe('ForgotPassword View', () => {
  it('Should render the forgot password page', () => {
    const { asFragment, getByTestId } = render(
      <Provider store={store}>
        <MemoryRouter>
          <ForgotPasswordView {...defaultProps} />
        </MemoryRouter>
      </Provider>,
    );
    expect(getByTestId('forgot-password-view')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
