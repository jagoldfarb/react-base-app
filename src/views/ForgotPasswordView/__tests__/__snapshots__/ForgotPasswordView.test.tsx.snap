// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword View Should render the forgot password page 1`] = `
<DocumentFragment>
  <div
    class="sc-gKsewC lbpPqi"
    data-testid="forgot-password-view"
  >
    <div
      class="sc-iBPRYJ jHiIsc"
    >
      <h1
        class="sc-fubCfw gKkqKR"
      >
        Forgot password
      </h1>
      <span
        class="sc-pFZIQ bXFDDZ"
      >
        Enter your email address to reset password.
      </span>
      <div
        class="sc-jrAGrp eRiotu"
      >
        <form
          class="sc-eCssSg jvctCF"
          data-testid="forgot-password-form"
        >
          <div
            class="sc-bdfBwQ cgTScR"
          >
            <div
              class="MuiFormControl-root MuiTextField-root sc-gsTCUz jKtOdX"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                data-shrink="false"
              >
                Email
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="input-Email"
                  maxlength="255"
                  name="email"
                  rows="3"
                  type="email"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-3"
                  >
                    <span>
                      Email
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="sc-jSgupP evxpKV"
          >
            <button
              class="sc-hKgILt bytOKm"
              data-testid="button"
              type="submit"
            >
              Reset my password
            </button>
          </div>
        </form>
      </div>
      <div
        class="sc-kEjbxe hAufN"
      >
        <a
          class="sc-iqHYGH hYDBwH"
          href="/signup"
        >
          Create account
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
